### ISSUE TITLE
metadata_expire change to Integer was a breaking change. String actually should be better

### ISSUE THREAD

**Issue by joemiller** [computology/packagecloud-cookbook#44]
This change broke our existing usage: https://github.com/computology/packagecloud-cookbook/commit/e379a98ed7194975e1e53952c29c276eccd205b7

Error we ran into:

```
================================================================================
Recipe Compile Error in /opt/pantheon/infrastructure/cookbooks/pantheon/recipes/default.rb
================================================================================

Chef::Exceptions::ValidationFailed
----------------------------------
Option metadata_expire must be a kind of Integer!  You passed "5m".

Cookbook Trace:
---------------
  /opt/pantheon/infrastructure/cookbooks/pantheon_package_repository/recipes/pantheon_repos.rb:11:in `block in from_file'
  /opt/pantheon/infrastructure/cookbooks/pantheon_package_repository/recipes/pantheon_repos.rb:8:in `from_file'
  /opt/pantheon/infrastructure/cookbooks/pantheon_package_repository/recipes/default.rb:17:in `block in from_file'
  /opt/pantheon/infrastructure/cookbooks/pantheon_package_repository/recipes/default.rb:12:in `each'
  /opt/pantheon/infrastructure/cookbooks/pantheon_package_repository/recipes/default.rb:12:in `from_file'
  /opt/pantheon/infrastructure/cookbooks/pantheon/recipes/default.rb:7:in `from_file'

Relevant File Content:
----------------------
/opt/pantheon/infrastructure/cookbooks/pantheon_package_repository/recipes/pantheon_repos.rb:

  4:  #
  5:  # Copyright (c) 2015 Pantheon Systems, Inc., All Rights Reserved.
  6:  #
  7:
  8:  packagecloud_repo 'pantheon/internal' do
  9:    type 'rpm'
 11>>   metadata_expire '5m'
 12:    priority 1
```

Since dnf and yum both allow for a string in the `metadata_expire` argument so that you can specify human-readable units, it seems like this cookbook should support the strings.

I am wondering if others involved w/ this cook agree with that or not? If so, the fix should be straightforward for v0.2.3. The default could be `'300'` (string) or `5m`. The validation regex that is in place would still be valid too.

**Comment by ice799** [comment]
ah yea that makes sense, my mistake. i'll put up a PR for this now.

**Comment by joemiller** [comment]
this is fixed now in 0.2.3. I'll close the issue.
